clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -68,6 +70,10 @@
             this.transport = transport;
         }
 
+        if (handshakeResponseTimeout != null) {
+            this.handshakeResponseTimeout = handshakeResponseTimeout;
+        }
+
         this.skipNegotiate = skipNegotiate;
 
         this.callback = (payload) -> {
```

Comment:
This logic is beginning to scare me. It looks quite a bit more complicated than the .NET logic. I think we'll need to do an "internals" review of the Java Client prior to RTM.